backend/server/routers/courses.py

```diff
@@ -301,9 +301,13 @@ def get_all_unlocked(userData: UserData) -> Dict[str, Dict]:
     """
 
     coursesState = {}
-    user = User(fix_user_data(userData.dict()))
+    user = User(fix_user_data(userData.model_dump()))
     for course, condition in CONDITIONS.items():
+        if course == "MATH1231":
+            print("MATH1231")
+            print("AAAAAAAAAAAAAAA")
         result, warnings = condition.validate(user) if condition is not None else (True, [])
+        print("ooop", result, warnings)
```

Comment:
```suggestion
        result, warnings = condition.validate(user) if condition is not None else (True, [])
```
was there something wrong with MATH1231 that should be further tested?

Reply:
It was a confusing error in autoplanning, but had nothing to do with this course code. Sorry, should have removed this.

Reply:
oh all good